echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy

```diff
@@ -102,6 +102,11 @@ class WebhooksController {
           sendEvent = false
         }
         log.info('Webhook event received {} {} {} {} {} {}', kv('type', type), kv('event_type', event.content.event_type), kv('hook_id', event.content.hook_id), kv('repository', event.content.repository.full_name), kv('request_id', event.content.request_id), kv('branch', event.content.branch))
+      } else if (source == 'gitlab') {
```

Comment:
Could you add a TODO & file an issue to replace this large `if else` block?

Reply:
ha! yeah, no problem!

Reply:
added!